src/detect_breakends/junction_detection.cpp

```diff
@@ -57,6 +66,26 @@ void detect_junctions_in_alignment_file(const std::filesystem::path & alignment_
             continue;
 
         const std::string ref_name = ref_ids[ref_id];
+        read_length = get_read_length(cigar);
+        // If there are positions to be sampled, then sample them.
+        if (!rand_sample.empty() && sample_size > 0)
+        {
+            if (ref_id == std::get<0>(rand_sample.front()) &&           // Correct chromosome.
+                pos <= std::get<1>(rand_sample.front()) &&              // Read starts before or at the position.
+                std::get<1>(rand_sample.front()) <= pos + read_length)  // Read ends after or at position.
+            {
+                ++cur_depth;
+            }
+            else if (ref_id > std::get<0>(rand_sample.front()) ||       // Record is at next chromosome
+                     (ref_id == std::get<0>(rand_sample.front()) &&     // Record is at same chromosome BUT
+                      pos > std::get<1>(rand_sample.front())))          // in a later position.
+            {
+                ++num_pos;
+                avg_depth += cur_depth;
+                cur_depth = 0;
+                rand_sample.pop_front();
+            }
+        }
```

Comment:
I'm not sure if it's such a good idea to put this into a separate function. I mean for one, it's basically just three if statements. The other thing is that I'm computing the depth at the current position, and then when we move to the next position this gets added to the average and then the current depth is zeroed out. So I'd have to pass all of these variables in and I think it would just get more confusing jumping back and forth between functions.

Reply:
My general feeling is that decoupling is always better and "just" 3 or 4 or 5 or 20 if branches can change pretty fast.